src/core/trtserver.h

```diff
@@ -492,6 +583,20 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestProviderNew(
     TRTSERVER_Server* server, const char* model_name, int64_t model_version,
     const char* request_header_base, size_t request_header_byte_size);
 
+/// Create a new inference request provider object. The caller retains ownership
+/// of 'request_options' but may release it by calling
+/// TRTSERVER_InferenceRequestOptionsDelete once this function returns.
+/// However, the values in 'request_options' must not be modified until the
```

Comment:
What does it mean that the options object can be delete but not its contents? For efficiency I think it is likely that we want to say that the provider does not take ownership but that the caller must extend the options object's lifetime to be longer than the lifetime of any provider created with the options. That way in the implementation we don't have to copy anything from options. But if we are copying everything from options anyway for other reasons then we can just say that the options object can be deleted immediately on return.